<!DOCTYPE html>
<html>
	<body>

		<h2>JavaScript Errors</h2>

		<h3>Async Await</h3>

		<p id="demo"></p>
		<p id="demo1"></p>

		<script>			
			function resolveAfter2Seconds(x) {
				return new Promise((resolve, reject) => {
					try {
						setTimeouttt(() => {//fix here: setTimeoutt -> setTimeout
							resolve(x*2);
						}, 1000);
					}catch {
						reject(TypeError("dasdasds"));
					}
				});
			}

			async function f1() {
					await resolveAfter2Seconds(10)
					.then((response) => {
						document.getElementById("demo").innerText = response;
						document.getElementById("demo1").innerText = "OlÃ¡";
					})
					.catch((err) => {
						document.getElementById("demo").innerText = err.name;
						document.getElementById("demo1").innerText = err.message;
					})
			}
			f1();
		</script>

	</body>
</html>
